test/cql-pytest/test_secondary_index.py

```diff
@@ -344,3 +346,104 @@ def test_index_weird_chars_in_col_name(cql, test_keyspace):
+        print(a)
+        cql.execute(*a)
+    def update_add(table, id, map, **kwargs):
+        a = (f'update {table} set m = m + %s where id = %s', (map, id))
+        print(a)
+        cql.execute(*a)
+    def delete(table, id, **kwargs):
+        a = (f'delete m from {table} where id = %s', (id,))
+        print(a)
+        cql.execute(*a)
+    def delete_cell(table, id, key, **kwargs):
+        a = (f'delete m[%s] from {table} where id = %s', (key, id))
+        print(a)
+        cql.execute(*a)
+
+    def random_map():
+        size = rand.randrange(len(possible_keys))
+        keys = rand.sample(possible_keys, k=size)
+        values = rand.choices(possible_values, k=size)
+        return dict(zip(keys, values))
+    def random_keys():
+        return set(random_map())
+    def random_key():
+        return random.choice(possible_keys)
+    def random_value():
+        return random.choice(possible_values)
+    def random_id():
+        return random.choice(possible_ids)
+
+    def random_operation():
+        return random.choice([insert, update_cell, update_delete, update_add, delete, delete_cell])
+    def random_args():
+        return {
+            'map': random_map(),
+            'key': random_key(),
+            'value': random_value(),
+            'keys': random_keys(),
+            'id': random_id(),
+        }
+
+    with new_test_table(cql, test_keyspace, schema) as tab1, new_test_table(cql, test_keyspace, schema) as tab2:
+        def select(cql, table, where, *args):
+            query = f'select id from {table} where {where}'
+            if table is tab2:
+                query += ' allow filtering'
+            try:
+                return cql.execute(query, *args)
+            except:
+                print('args=', args, table, where)
+                raise
+        def test_all():
+            possible_ids_ = possible_ids + [10000]
+            possible_keys_ = possible_keys + [10000]
+            possible_values_ = possible_values + ['aaaaa']
+            for k, v in itertools.product(possible_keys_, possible_values_):
+                r1 = select(cql, tab1, 'm[%s] = %s', (k, v))
+                r2 = select(cql, tab2, 'm[%s] = %s', (k, v))
+                assert_rows_ignoring_order(r1, *list(r2))
+            for k in possible_keys_:
+                r1 = select(cql, tab1, 'm contains key %s', (k,))
+                r2 = select(cql, tab2, 'm contains key %s', (k,))
+                assert_rows_ignoring_order(r1, *list(r2))
+            for v in possible_values_:
+                r1 = select(cql, tab1, 'm contains %s', (v,))
+                r2 = select(cql, tab2, 'm contains %s', (v,))
+                assert_rows_ignoring_order(r1, *list(r2))
+
+
+        cql.execute(f'create index on {tab1}(keys(m))')
+        cql.execute(f'create index on {tab1}(values(m))')
+        cql.execute(f'create index on {tab1}(entries(m))')
+
+        for _ in range(1000):
```

Comment:
How slow is this test with 1000 iterations? Maybe a lower number would have been enough?

Reply:
It's too long admittedly. Took ~30 seconds on dev compilation of Scylla. Maybe I'll lower it to ~50 and lower the possible ids, keys and values that I insert.

Reply:
Issue is that the code is a bit brittle, as it involves many if-elses regarding where to source from the liveness info for resulting rows. 1000 iterations is excessive, but 50 should be passable. See updated test.